renderdoc/api/replay/vk_pipestate.h

```diff
@@ -596,6 +596,15 @@ value.
 :type: bytes
 )");
   bytebuf specializationData;
+
+  // NOTE: This mirrors patchData->specIDs.
+  DOCUMENT(R"(Specialization constant indices for each entry in specializationData, representing
+the specialization constant layout. Element at `specializationData[i*8]` refers to specialization
+constant `specializationIds[i]`.
```

Comment:
It feels a bit wrong needing to 'leak' this information to the UI as it shouldn't be needed, but indeed I can't think of a cleaner solution while still generating the fossilize DB in the UI. Since constant IDs are no longer used directly as an offset (unfortunately necessary as some applications use the full 32-bit range) they can no longer be implicit so some correlation is needed.

Storing the spec ID on every constant in the reflection info like the default value is too much (and I am strongly considering removing that default value from there in future refactors). Saving and sharing the 'unmodified' specialisation buffer just for this purpose is overkill and likely to cause confusion as to which buffer to use since they're arranged differently.

The only thing I'd suggest is rewording the comment here. The data is still the same either way, but this array should be described in terms of its relationship to the list of shader constants in the specialisation constant `ConstantBlock`. In the same order, giving their API constant IDs. This then allows correlation between a constant ID and an offset by going via the shader constant for its `byteOffset` as noted above. Avoid making a direct comparison between the IDs and where they're stored in case the packing needs to change again in future.